gen/golden_test.go

```diff
@@ -0,0 +1,67 @@
+package gen
+
+import (
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func hash(t *testing.T, name string) string {
+	f, err := os.Open(name)
+	defer f.Close()
+	require.NoError(t, err)
+
+	h := sha1.New()
+	_, err = io.Copy(h, f)
+	require.NoError(t, err)
+
+	return fmt.Sprintf("% x", h.Sum(nil))
+}
+
+func run(name string, args ...string) error {
+	cmd := exec.Command(name, args...)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func TestCodeIsUpToDate(t *testing.T) {
+	// This test just verifies that the generated code in testdata/ is up to
+	// date. If this test failed, run 'make generate' in the testdata/ directory
+	// and commit the changes.
+
+	files, err := filepath.Glob("testdata/*.thrift")
+	require.NoError(t, err)
+
+	outputDir := filepath.Join(os.TempDir(), "testdata")
```

Comment:
prefer if we used `ioutil.TempDir` here, makes the test resilient to having multiple copies run at the same time
```go
outputDir := ioutil.TempDir("", "testdata")
```